packages/attune/src/server/repo/index/sign.rs

```diff
@@ -966,6 +905,735 @@ pub async fn handler(
         }
     }
+}
 
-    Ok(Json(SignIndexResponse {}))
+#[cfg(test)]
+mod tests {
+    use std::iter::once;
+
+    use async_tempfile::TempDir;
+    use axum_test::multipart::{MultipartForm, Part};
+    use gpgme::{Context, CreateKeyFlags, ExportMode, Protocol};
+
+    use super::*;
+    use crate::{
+        server::{
+            pkg::upload::PackageUploadResponse,
+            repo::{
+                create::repo_prefix,
+                index::generate::{GenerateIndexRequest, GenerateIndexResponse},
+                sync::check::CheckConsistencyResponse,
+            },
+        },
+        testing::{AttuneTestServer, AttuneTestServerConfig},
+    };
+
+    const TEST_PACKAGE_AMD64: &[u8] =
+        include_bytes!("./fixtures/attune-test-package_2.0.0_linux_amd64.deb");
+    const TEST_PACKAGE_ARM64: &[u8] =
+        include_bytes!("./fixtures/attune-test-package_2.0.0_linux_arm64.deb");
+
+    async fn sign_index(index: &str) -> (String, String, String) {
+        let dir = TempDir::new().await.unwrap();
+        let mut gpg = Context::from_protocol(Protocol::OpenPgp).unwrap();
+        gpg.set_engine_home_dir(dir.dir_path().to_str().unwrap())
+            .unwrap();
+        gpg.set_armor(true);
+        let keygen_result = gpg
+            .create_key_with_flags(
+                "Attune Test",
+                "default",
+                Default::default(),
+                CreateKeyFlags::NOPASSWD,
+            )
+            .unwrap();
+        let key_id = keygen_result.fingerprint().unwrap();
+        let key = gpg
+            .find_secret_keys(vec![key_id])
+            .unwrap()
+            .next()
+            .unwrap()
+            .unwrap();
+        gpg.add_signer(&key).unwrap();
+
+        let mut clearsigned = Vec::new();
+        gpg.sign_clear(index.as_bytes(), &mut clearsigned)
+            .expect("could not clearsign index");
+        let clearsigned =
+            String::from_utf8(clearsigned).expect("clearsigned index contained invalid characters");
+        debug!(?index, ?clearsigned, "clearsigned index");
+        let mut detachsigned = Vec::new();
+        gpg.sign_detached(index.as_bytes(), &mut detachsigned)
+            .expect("could not detach sign index");
+        let detachsigned = String::from_utf8(detachsigned)
+            .expect("detachsigned index contained invalid characters");
+        debug!(?index, ?detachsigned, "detachsigned index");
+
+        let mut public_key_cert = Vec::new();
+        gpg.export_keys(once(&key), ExportMode::empty(), &mut public_key_cert)
+            .expect("could not export key");
+        let public_key_cert = String::from_utf8(public_key_cert)
+            .expect("public key cert contained invalid characters");
+        debug!(?public_key_cert, "public key cert");
+
+        (clearsigned, detachsigned, public_key_cert)
+    }
+
+    #[sqlx::test(migrator = "crate::testing::MIGRATOR")]
+    #[test_log::test]
```

Comment:
huh, interesting that sqlx is one of the ones that works here.
for reference, usually i just wrap these:
```rs
#[#[test_log::test(sqlx::test(migrator = "crate::testing::MIGRATOR"))]
```

but of course doing it this way is fine too.